### PR TITLE
Fix path separator lookup in Unicode paths on Windows

### PR DESCRIPTION
**Pull request by mloskot**
Apparently, searching UTF8-encoded string with `std::string::find*` routines may fail to return correct position for some character sets.
Thus, search for path separator in UTF-16 string.

Follows up PR #2135 and refines PR #2160.

-----

For example: for path `F:\tmp\PDAL\日本語\06LC743.las` this `size_t found = path.find_last_of("/\\");` yields value `21` and leading later to composition of broken path `F:\tmp\PDAL\日本語\06LC7\06LC743.las`, while the correct value of `found` is `15`.

### CODE REVIEW

**Comment by mloskot** [comment]
Looks like the AppVeyor failure is not related to this change. It's already failed in https://ci.appveyor.com/project/hobu/pdal/builds/19515768 following @connormanning -s via @abellgithub commit https://github.com/PDAL/PDAL/commit/8cfa74929831ddea8080bd5a6b64dd36d238374e

**Comment by abellgithub** [comment]
Was this a real issue that needs to be addressed or just a speculative fix for some other issue you saw?  If the latter, I'd just assume close without change.

**Comment by mloskot** [comment]
@abellgithub 
> Was this a real issue that needs to be addressed or just a speculative fix for some other issue you saw?

What are you referring to? The AppVeyor's issues or goal behind this PR?

**Comment by abellgithub** [comment]
I misunderstood what was going on when you made a comment about Appveyor.  Looking again, it seems like this change was unrelated to that.  Correct?

**Comment by mloskot** [comment]
Yes, this change is unrelated to the AppVeyor failures.
This PR just fixes a bug in Unicode filesystem paths support on Windows.

**Comment by abellgithub** [comment]
Great.  Can you add/modify a test that exercises this code?

**Comment by mloskot** [comment]
@abellgithub I've update the `test/unit/FileUtilsTest.cpp` to cover the case explained above (ie. path like `F:\tmp\PDAL\日本語\06LC743.las`). I've also updated directory operations to convert the input path name to `std::wstring` on Windows. 

I've tested this PR on Linux using GCC 5.5.
On Windows, I'll have to rely on AppVeyor for today, temporarily.

File: test/unit/FileUtilsTest.cpp [hunk 1]
```diff
@@ -39,6 +39,8 @@
 
 #include "Support.hpp"
 
+#include <iostream>
```
Comment (mloskot) [inline test/unit/FileUtilsTest.cpp]: Grrr, I should have removed it before updating the commit.
I will wait for the CI-s first, then remove it.

**Comment by mloskot** [comment]
The test is failing on Windows - https://ci.appveyor.com/project/hobu/pdal/builds/19587690

```
[00:43:18] 10: Test command: C:\projects\pdal\build\bin\pdal_file_utils_test.exe
[00:43:18] 10: Environment variables: 
[00:43:18] 10:  PDAL_DRIVER_PATH=C:/projects/pdal/build/bin
[00:43:18] 10: Test timeout computed to be: 10000000
[00:43:18] 10: [==========] Running 10 tests from 1 test case.
[00:43:18] 10: [----------] Global test environment set-up.
[00:43:18] 10: [----------] 10 tests from FileUtilsTest
[00:43:18] 10: [ RUN      ] FileUtilsTest.test_file_ops
[00:43:18] 10: [       OK ] FileUtilsTest.test_file_ops (5 ms)
[00:43:18] 10: [ RUN      ] FileUtilsTest.test_readFileIntoString
[00:43:18] 10: [       OK ] FileUtilsTest.test_readFileIntoString (2 ms)
[00:43:18] 10: [ RUN      ] FileUtilsTest.test_toAbsolutePath
[00:43:18] 10: [       OK ] FileUtilsTest.test_toAbsolutePath (1 ms)
[00:43:18] 10: [ RUN      ] FileUtilsTest.test_getDirectory
[00:43:18] 10: [       OK ] FileUtilsTest.test_getDirectory (0 ms)
[00:43:18] 10: [ RUN      ] FileUtilsTest.test_isAbsolute
[00:43:18] 10: [       OK ] FileUtilsTest.test_isAbsolute (0 ms)
[00:43:18] 10: [ RUN      ] FileUtilsTest.filename
[00:43:18] 10: [       OK ] FileUtilsTest.filename (0 ms)
[00:43:18] 10: [ RUN      ] FileUtilsTest.extension
[00:43:18] 10: [       OK ] FileUtilsTest.extension (0 ms)
[00:43:18] 10: [ RUN      ] FileUtilsTest.stem
[00:43:18] 10: [       OK ] FileUtilsTest.stem (0 ms)
[00:43:18] 10: [ RUN      ] FileUtilsTest.glob
[00:43:18] 10: [       OK ] FileUtilsTest.glob (5 ms)
[00:43:18] 10: [ RUN      ] FileUtilsTest.test_file_ops_with_unicode_paths
[00:43:18] 10: C:\projects\pdal\test\unit\FileUtilsTest.cpp(296): error: Expected: (std::count(filenames.cbegin(), filenames.cend(), tmp1)) > (0), actual: 0 vs 0
[00:43:18] 10: C:\projects\pdal\test\unit\FileUtilsTest.cpp(296): error: Expected: (std::count(filenames.cbegin(), filenames.cend(), tmp1)) > (0), actual: 0 vs 0
[00:43:18] 10: [  FAILED  ] FileUtilsTest.test_file_ops_with_unicode_paths (6 ms)
[00:43:18] 10: [----------] 10 tests from FileUtilsTest (19 ms total)
[00:43:18] 10: 
[00:43:18] 10: [----------] Global test environment tear-down
[00:43:18] 10: [==========] 10 tests from 1 test case ran. (19 ms total)
[00:43:18] 10: [  PASSED  ] 9 tests.
[00:43:18] 10: [  FAILED  ] 1 test, listed below:
[00:43:18] 10: [  FAILED  ] FileUtilsTest.test_file_ops_with_unicode_paths
[00:43:18] 10: 
[00:43:18] 10:  1 FAILED TEST
[00:43:18] 10/95 Test #10: pdal_file_utils_test ...................***Failed    0.06 sec
```

I will investigate it as soon as possible.

**Comment by mloskot** [comment]
@abellgithub  This is ready to merge

Failure of most recent build of this PR on https://ci.appveyor.com/project/hobu/pdal/builds/19634833 is not related to this PR. The `FileUtilsTest.cpp` cases all pass:

```
[00:36:43] 10: [==========] Running 10 tests from 1 test case.
[00:36:43] 10: [----------] Global test environment set-up.
[00:36:43] 10: [----------] 10 tests from FileUtilsTest
[00:36:43] 10: [ RUN      ] FileUtilsTest.test_file_ops
[00:36:43] 10: [       OK ] FileUtilsTest.test_file_ops (4 ms)
[00:36:43] 10: [ RUN      ] FileUtilsTest.test_readFileIntoString
[00:36:43] 10: [       OK ] FileUtilsTest.test_readFileIntoString (2 ms)
[00:36:43] 10: [ RUN      ] FileUtilsTest.test_toAbsolutePath
[00:36:43] 10: [       OK ] FileUtilsTest.test_toAbsolutePath (0 ms)
[00:36:43] 10: [ RUN      ] FileUtilsTest.test_getDirectory
[00:36:43] 10: [       OK ] FileUtilsTest.test_getDirectory (1 ms)
[00:36:43] 10: [ RUN      ] FileUtilsTest.test_isAbsolute
[00:36:43] 10: [       OK ] FileUtilsTest.test_isAbsolute (0 ms)
[00:36:43] 10: [ RUN      ] FileUtilsTest.filename
[00:36:43] 10: [       OK ] FileUtilsTest.filename (0 ms)
[00:36:43] 10: [ RUN      ] FileUtilsTest.extension
[00:36:43] 10: [       OK ] FileUtilsTest.extension (0 ms)
[00:36:43] 10: [ RUN      ] FileUtilsTest.stem
[00:36:43] 10: [       OK ] FileUtilsTest.stem (0 ms)
[00:36:43] 10: [ RUN      ] FileUtilsTest.glob
[00:36:43] 10: [       OK ] FileUtilsTest.glob (5 ms)
[00:36:43] 10: [ RUN      ] FileUtilsTest.test_file_ops_with_unicode_paths
[00:36:43] 10: [       OK ] FileUtilsTest.test_file_ops_with_unicode_paths (6 ms)
[00:36:43] 10: [----------] 10 tests from FileUtilsTest (18 ms total)
[00:36:43] 10: 
[00:36:43] 10: [----------] Global test environment tear-down
[00:36:43] 10: [==========] 10 tests from 1 test case ran. (18 ms total)
[00:36:43] 10: [  PASSED  ] 10 tests.
[00:36:43] 10/95 Test #10: pdal_file_utils_test ...................   Passed  
```

**Pull request merged**